src/operator/race-static.ts

```diff
@@ -8,14 +8,14 @@ import {isArray} from '../util/isArray';
  * @param {...Observables} ...observables sources used to race for which Observable emits first.
  * @returns {Observable} an Observable that mirrors the output of the first Observable to emit an item.
  */
-export function race<T>(...observables: Array<Observable<any> | Array<Observable<any>>>): Observable<T> {
+export function race<T>(...observables: Array<Observable<T> | Array<Observable<T>>>): Observable<T> {
   // if the only argument is an array, it was most likely called with
   // `pair([obs1, obs2, ...])`
   if (observables.length === 1) {
-    if (isArray(observables[0])) {
-      observables = <Array<Observable<any>>>observables[0];
-    } else {
+    if (!isArray(observables[0])) {
```

Comment:
I'm generally in favor of pattern 
```
if () {
 return; //return early if condition met(or not met)
}
```

but in this flow is there strong differences compare previous one? Either case this operator returns only when `observable[0]` is array and only order seems been changed.